docs/how-to/h-cluster-migration.md

```diff
@@ -0,0 +1,171 @@
+# Cluster Migration using MirrorMaker2.0
+
+## Overview
+
+This How-To guide covers executing a cluster migration to a Charmed Kafka K8s deployment using MirrorMaker2.0, running as a process on each of the Juju units in an active/passive setup, where MirrorMaker will act as a consumer from an existing cluster, and a producer to the Charmed Kafka K8s cluster. In parallel (one process on each unit), data and consumer offsets for all existing topics will be synced one-way until both clusters are in-sync, with all data replicated across both in real-time.
+
+## MirrorMaker2 Overview
+
+Under the hood, MirrorMaker uses Kafka Connect source connectors to replicate data, those being the following:
+- **MirrorSourceConnector** - replicates topics from an original cluster to a new cluster. It also replicates ACLs and is necessary for the MirrorCheckpointConnector to run
+- **MirrorCheckpointConnector** - periodically tracks offsets. If enabled, it also synchronizes consumer group offsets between the original and new clusters
+- **MirrorHeartbeatConnector** - periodically checks connectivity between the original and new clusters
+
+Together, these allow for cluster->cluster replication of topics, consumer groups, topic configuration and ACLs, preserving partitioning and consumer offsets. For more detail on MirrorMaker internals, consult the [MirrorMaker README.md](https://github.com/apache/kafka/blob/trunk/connect/mirror/README.md) and the [MirrorMaker 2.0 KIP](https://cwiki.apache.org/confluence/display/KAFKA/KIP-382%3A+MirrorMaker+2.0). In practice, it allows one to sync data one-way between two live Kafka clusters with minimal impact on the ongoing production service.
+
+In short, MirrorMaker runs as a distributed service on the new cluster, and consumes all topics, groups and offsets from the still-active original cluster in production, before producing them one-way to the new cluster that may not yet be serving traffic to external clients. The original, in-production cluster is referred to as an ‘active’ cluster, and the new cluster still waiting to serve external clients is ‘passive’. The MirrorMaker service can be configured using much the same configuration as available for Kafka Connect.
+
+## Deployment and Running of MirrorMaker
+
+### Pre-requisites
+
+- An existing Kafka cluster to migrate from. The clusters need to be reachable from/to Charmed Kafka K8s. 
+- A bootstrapped Juju K8s cloud running Charmed Kafka K8s to migrate to
+    - A tutorial on how to set-up a Charmed Kafka deployment can be found as part of the [Charmed Kafka K8s Tutorial](/t/charmed-kafka-k8s-documentation-tutorial-overview/11945)
+    - `yq` - https://github.com/mikefarah/yq
+    - `snap install yq --channel=v3/stable`
+
+### Getting Charmed Kafka K8s cluster details and admin credentials
```

Comment:
```suggestion
### Get cluster details and admin credentials
```
(Unless you think it's not clear enough that we are referring to the Charmed Kafka K8 cluster specifically)